onnxruntime/core/framework/session_state.cc

```diff
@@ -269,14 +269,64 @@ Status SessionState::GeneratePatternGroupCache(const std::vector<std::reference_
   ORT_ENFORCE(exe_plan);
   OrtValuePatternPlanner mem_planner(*exe_plan);
   auto& node_index_info = GetNodeIndexInfo();
+
+  // Contigiously allocate activations.
+  // TODO(codemzs): Refacor this code.
+  for (auto ml_value_idx : exe_plan->activation_allocation_order) {
+
+    const auto* ml_type = exe_plan->allocation_plan[ml_value_idx].value_type;
+    if (!ml_type->IsTensorType())
+      continue;
+    const auto* ml_data_type = static_cast<const TensorTypeBase*>(ml_type)->GetElementType();
+    
+    // TODO(codemzs): Assert everything is kAllocate.
+
+    if (exe_plan->allocation_plan[ml_value_idx].alloc_kind == AllocKind::kAllocate &&
+        ml_data_type != DataTypeImpl::GetType<std::string>()) {
+      //calculate size
+      auto* arg = exe_plan->allocation_plan[ml_value_idx].p_def_site;
+      if (!arg->Shape())
+        continue;
```

Comment:
Does this mean that if a tensor doesn't have a shape (e.g. shape inferencing failed), then we will not allocate it contiguously with the rest?  I'm wondering if we need to fail here, instead of skip it, or does this get checked somewhere else?

Reply:
Thanks, my intent was to FAIL here, this was an oversight.